modules/ppcp-button/resources/js/modules/Renderer/MessageRenderer.js

```diff
@@ -30,17 +33,36 @@ class MessageRenderer {
             return;
         }
 
+        const options = {
+            amount,
+            placement: this.config.placement,
+            style: this.config.style
+        };
+
+        if (this.isOptionsFingerprintEqual(options)) {
+            return;
+        }
+
         const newWrapper = document.createElement('div');
         newWrapper.setAttribute('id', this.config.wrapper.replace('#', ''));
 
-        const sibling = document.querySelector(this.config.wrapper).nextSibling;
-        document.querySelector(this.config.wrapper).parentElement.removeChild(document.querySelector(this.config.wrapper));
+        const oldWrapper = document.querySelector(this.config.wrapper);
+        const sibling = oldWrapper.nextSibling;
+        oldWrapper.parentElement.removeChild(oldWrapper);
         sibling.parentElement.insertBefore(newWrapper, sibling);
-        paypal.Messages({
-            amount,
-            placement: this.config.placement,
-            style: this.config.style
-        }).render(this.config.wrapper);
+
+        paypal.Messages(options).render(this.config.wrapper);
+    }
+
+    isOptionsFingerprintEqual(options) {
```

Comment:
I would just call it something like `optionsEqual` without the fingerprint word, seems unneeded and can be confusing.

Reply:
I agree, will change.